sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/UserGroups.java

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.spring.autoconfigure.aad;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class UserGroups {
+
+    @JsonProperty("odata.nextLink")
+    private Optional<String> odataNextLink;
```

Comment:
We can set default value to `Optional.empty()` to avoid `NullPointerException`.
Refs: https://stackoverflow.com/questions/18805455/setting-default-values-to-null-fields-when-mapping-with-jackson

But `having an Optional in a class field or in a data structure, is considered a misuse of the API`,
Refs: https://stackoverflow.com/questions/23454952/uses-for-optional

So I suggest to use
```
private String odataNextLink;
```
instead of 
```
private Optional<String> odataNextLink;
```

Reply:
Thanks for pointing this out, using a `String` is the better approach! I made the change in https://github.com/Azure/azure-sdk-for-java/pull/14305/commits/cc20c0521dbed11c5ffc661c23de8ed1ff65e9b4